src/services/DispatchService.ts

```diff
@@ -27,39 +27,46 @@ class DispatchService {
 
   public async processSQSRecord(record: SQSRecord): Promise<void> {
     const target: Target = getTargetFromSourceARN(record.eventSourceARN);
-    const body: Body = JSON.parse(record.body) as Body;
-    const removeEvent = body.eventType === 'REMOVE';
-    const targetRecord: TargetRecord = {
-      eventType: body.eventType,
-      body: undefined,
-    };
-    if (!['INSERT', 'MODIFY', 'REMOVE'].includes(body.eventType)) {
+    const dynamoEvent: DynamoDBRecord = JSON.parse(record.body) as DynamoDBRecord;
+
+    if (!dynamoEvent.eventName) {
+      this.logger.error("no event name present");
+      this.logger.error(ERROR.FAILED_VALIDATION_SENDING_TO_DLQ);
+      await this.sendRecordToDLQ(record.body, target);
+    }
+
+    if (!['INSERT', 'MODIFY', 'REMOVE'].includes(dynamoEvent.eventName!)) {
+      this.logger.error("not a valid event name");
       this.logger.error(ERROR.FAILED_VALIDATION_SENDING_TO_DLQ);
       await this.sendRecordToDLQ(record.body, target);
       return;
     }
-    if (!removeEvent) {
-      if (!body.body.NewImage) {
-        this.logger.error(ERROR.NO_NEW_IMAGE);
-        await this.sendRecordToDLQ(record.body, target);
-        return;
-      }
-      targetRecord.body = DynamoDB.Converter.unmarshall(body.body.NewImage);
+
+    if (!dynamoEvent.dynamodb?.NewImage) {
```

Comment:
I'm pretty sure that REMOVE events don't contain a new NewImage.  Will this ship all events that don't contain a NewImage to the DLQ?

Reply:
Addressed: https://github.com/dvsa/cvs-tsk-edh-dispatcher/pull/10/commits/36de114229059d8a15776af8eb0f10730cf073a0